src/main/solution_space/path_finder/path_finder_v_3.py

```diff
@@ -126,27 +139,27 @@ def __is_most_of_path_is_done(diffs_from_empty_to_goal: int, diffs_from_user_to_
         return diffs_from_user_to_goal <= diffs_from_empty_to_goal * DIFFS_PERCENT_TO_GO_DIRECTLY
 
     @staticmethod
-    def __is_rate_worse(user_rate: float, graph_vertex_rate: float):
+    def __is_rate_worse(user_rate: float, graph_vertex_rate: float) -> bool:
         # TODO: 14/04 or If number of passed tests > 0 then True???
         return user_rate > 0 and graph_vertex_rate == 0
 
     # Returns should we go through graph or directly to the goal
-    def __go_through_graph(self, user_vertex: Vertex, graph_vertex: Vertex, goal: Vertex) -> bool:
+    def __go_through_graph(self, user_anon: AnonTree, graph_anon: AnonTree, goal_anon: AnonTree) -> bool:
         """
         1. If __is_most_of_path_is_done, return False
         2. If __is_rate_worse, return False
         3. Return not __is_far_from_graph
         """
-        diffs_from_user_to_goal = user_vertex.get_dist(goal)
-        diffs_from_empty_to_user = self._graph.empty_vertex.get_dist(user_vertex)
+        empty_anon = self._graph.empty_vertex.serialized_code.anon_trees[0]
+        # Todo: use anon files after merging
+        diffs_from_user_to_goal = GumTreeDiff.create_tmp_files_and_get_diffs_number(user_anon.tree, goal_anon.tree)
```

Comment:
Anon tree has files, but it is in my PR. A think we can merge my PR and you will fix it. Because we will spend additional time for creating tmp files. What do you think?

Reply:
Yeah, this is why I wrote the comment `# Todo: use anon files after merging`